myserver.cpp

```diff
@@ -0,0 +1,553 @@
+    switch(iRtCode){
+    case SUCCESS:
+        socketWrite("SUCCESS");
+        break;
+    case NO_SUCH_USER:
+        socketWrite("NO_SUCH_USER");
+        break;
+    case NO_SUCH_CATEGORY:
+        socketWrite("NO_SUCH_CATEGORY");
+        break;
+    case UNEXPECTED_ERROR:
+        socketWrite("UNEXPECTED_ERROR");
+        break;
+    case INVALID_INPUT_DATA:
+        socketWrite("INVALID_INPUT_DATA");
+        break;
+    }
+}
+void CMyServer::AddNote(const std::vector <std::string> &msg){
+    std::string sKeyWords;
+    iNoteId = std::stoi(msg.at(1));
+    iCategoryId = std::stoi(msg.at(2));
+    sKeyWords = msg.at(3);
+    int iRtCode = IDataBase::AddNote(iCategoryId, sKeyWords, iNoteId);
+    QJsonObject AddNote;
+    switch(iRtCode){
+    case SUCCESS:
+        AddNote.insert("RequestType", "AddNote");
+        AddNote.insert("SUCCESS", iNoteId);
+        document.setObject(AddNote);
+        bytes = document.toJson( QJsonDocument::Indented );
+        socketWrite(bytes);
+        break;
+    case INVALID_INPUT_DATA:
+        socketWrite("INVALID_INPUT_DATA");
+        break;
+    case NO_SUCH_CATEGORY:
+        socketWrite("NO_SUCH_CATEGORY");
+        break;
+    case UNEXPECTED_ERROR:
+        socketWrite("UNEXPECTED_ERROR");
+        break;
+    }
+}
+void CMyServer::GetNote(const std::vector <std::string> &msg){
+    std::vector<std::string> vsNote;
+    int iRtCode = IDataBase::GetNote(iNoteId, iUserId, vsNote);
+    QJsonObject GetNote;
+    std::string sGNote;
+    switch(iRtCode){
+    case SUCCESS:
+        for(int i = 0; i <= vsNote.size(); i++){
+             sGNote += vsNote[i] + '|';
+        }
+        GetNote.insert("RequestType", "GetNote");
+        GetNote.insert("SUCCESS", sGNote.data());
+        document.setObject(GetNote);
+        bytes = document.toJson( QJsonDocument::Indented );
+        socketWrite(bytes);
+        break;
+    case NO_SUCH_USER:
+        socketWrite("NO_SUCH_USER");
+        break;
+    case NO_SUCH_NOTE:
+        socketWrite("NO_SUCH_NOTE");
+        break;
+    case UNEXPECTED_ERROR:
+        socketWrite("UNEXPECTED_ERROR");
+        break;
+    }
+}
+void CMyServer::ChangeCategoryId(const std::vector <std::string> &msg){
+    iNoteId = std::stoi(msg.at(1));
+    iCategoryId = std::stoi(msg.at(2));
+    iUserId = std::stoi(msg.at(3));
+    int iRtCode = IDataBase::ChangeCategoryId(iNoteId, iCategoryId, iUserId);
+    switch(iRtCode){
+    case SUCCESS:
+        socketWrite("SUCCESS");
+        break;
+    case NO_SUCH_USER:
+        socketWrite("NO_SUCH_USER");
+        break;
+    case NO_SUCH_NOTE:
+        socketWrite("NO_SUCH_NOTE");
+        break;
+    case NO_SUCH_CATEGORY:
+        socketWrite("NO_SUCH_CATEGORY");
+        break;
+    case UNEXPECTED_ERROR:
+        socketWrite("UNEXPECTED_ERROR");
+        break;
+    }
+}
+void CMyServer::SetHeader(const std::vector <std::string> &msg){
+    std::string sHeader;
+    iNoteId = std::stoi(msg.at(1));
```

Comment:
Same here, if `msg` had been empty, this would cause exception and your server would crash

Reply:
Fixed